src/book.py

```diff
@@ -574,73 +577,79 @@ def _read_kraken_ledgers(self, file_path: Path) -> None:
+                        )
+                        raise RuntimeError
+                    # This should never happen
+                    else:
+                        log.error(
+                            f"{file_path} row {row}: Unknown value for appended "
+                            f"operation flag {self.kraken_held_ops[refid]['appended']}."
+                            "Please create an Issue or PR."
+                        )
+                        raise RuntimeError
 
-                    if operation == "Deposit":
-                        # append only the second deposit to the operations list
-                        append_operation = len(refid_idxs) == 1
-                    elif operation == "Withdrawal":
-                        # Append the first withdrawal to the operations list as soon
-                        # as the second withdrawal occurs. Therefore, for the second
-                        # withdrawal, overwrite the variables with the data of the first
-                        # withdrawal and append this to the operations list.
-                        if refid_idxs:
-                            append_operation = True
-                            # required for type annotation: convert Optional[str] to str
-                            operation = str(held_operations[idx]["operation"])
-                            utc_time = held_operations[idx]["utc_time"]
-                            platform = held_operations[idx]["platform"]
-                            change = held_operations[idx]["change"]
-                            coin = held_operations[idx]["coin"]
-                            row = held_operations[idx]["row"]
-                            file_path = held_operations[idx]["file_path"]
-                            fee = held_operations[idx]["fee"]
-                        else:
-                            append_operation = False
+                    # For deposits, this is all we need to do.
+                    # For withdrawals, we need to append the first withdrawal as soon as
+                    # the second withdrawal occurs. Therefore, overwrite the variables
+                    # with the data of the first withdrawal and append it.
+                    if append_operation and operation == "Withdrawal":
+                        # required for type annotation: convert Optional[str] to str
+                        operation = str(self.kraken_held_ops[refid]["operation"])
+                        utc_time = self.kraken_held_ops[refid]["utc_time"]
+                        platform = self.kraken_held_ops[refid]["platform"]
+                        change = self.kraken_held_ops[refid]["change"]
+                        coin = self.kraken_held_ops[refid]["coin"]
+                        row = self.kraken_held_ops[refid]["row"]
+                        file_path = self.kraken_held_ops[refid]["file_path"]
+                        fee = self.kraken_held_ops[refid]["fee"]
```

Comment:
If we use tuples to store the operation data and tuple unpacking (`a, b =tuple([1,2])`), we can make sure, that we do not forget to update this line of code in the future, when we add more operation data. 

Btw why don't we store the operation as real operation class? I think that it would look a lot nicer instead of always passing each value in it's own. 

We have to create a new help function for this, which creates the only the operation class an does not add it to the list of operations directly. So we call `create_operation` and use `self.operstions.append(op)` when we want to add the stored operation. 

What do think about that?

Reply:
Hmm, it sounds cleaner but I don't know if it really makes things easier. If we add more items to the operation in the future, we have to revisit this place anyways (be it for append_operation or create_operation). Additionally, we would need to store an operation for the fee as well, effectively needing twice the memory due to the overhead of storing timestamp, platform etc. again. Right now, the fee is just an additional number. Furthermore, we would need an additional logic to decide between deposit/withdrawal and any other operation when actually appending the operation.

Reply:
We already have the logic to determine the operation type. Its the same that append operation uses

Reply:
Yeah we need to reserve more memory, but I find the implementation way more clearer, that I find it worth